gobblin-yarn/src/main/java/gobblin/yarn/YarnHelixUtils.java

```diff
@@ -199,4 +206,48 @@ public static void addFileAsLocalResource(FileSystem fs, Path destFilePath, Loca
 
     return environmentVariableMap;
   }
-}
+
+  public static Optional<Integer> takePort(int desiredPort) {
```

Comment:
I can add a minimum bounds check.  I also considered adding a optional maximum to the range.  Lastly, I might be wrong, but I thought that only the first ~1000 ports where “reserved”.